server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -609,7 +621,9 @@ static void buildShardLevelInfo(
             }
             final IndexingStats indexingStats = s.getStats().getIndexing();
             if (indexingStats != null) {
-                final double shardWriteLoad = indexingStats.getTotal().getPeakWriteLoad();
+                final double shardWriteLoad = (shardWriteLoadType == WriteLoadDeciderShardWriteLoadType.PEAK)
+                    ? indexingStats.getTotal().getPeakWriteLoad()
+                    : indexingStats.getTotal().getRecentWriteLoad();
```

Comment:
Nit: we could maybe implement

```java
public double getWriteLoad(IndexingStats indexingStats) {
   // ...
}
```
on `WriteLoadDeciderShardWriteLoadType` and do
```java
final double shardWriteLoad = shardWriteLoadType.getWriteLoad(indexingStats)
```
 here?